elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,375 @@
+        header_right_column = pl.get_string_attrib(element, 'header-right-column', 'Construct your solution here:')
+
+        student_submission_dict_list = []
+
+        mcq_options = data['params'][answer_name]
+
+        if answer_name in data['submitted_answers']:
+            student_previous_submission = data['submitted_answers'][answer_name]['student_raw_submission']
+            mcq_options = list(set(mcq_options) - set(student_previous_submission))
+
+        for index, mcq_options_text in enumerate(student_previous_submission):
+            # render the answers column (restore the student submission)
+            submission_indent = data['submitted_answers'][answer_name]['student_answer_indent'][index]
+            submission_indent = (int(submission_indent) * 50) + 10
+            temp = {'text': mcq_options_text, 'indent': submission_indent}
+            student_submission_dict_list.append(dict(temp))
+
+        dropzone_layout = pl.get_string_attrib(element, 'dropzone-layout', 'horizontalLayout')
+
+        html_params = {
+            'question': True,
+            'answer_name': answer_name,
+            'options': mcq_options,
+            'header-left-column': header_left_column,
+            'header-right-column': header_right_column,
+            'submission_dict': student_submission_dict_list,
+            'dropzone_layout': 'verticalLayout' if dropzone_layout == 'verticalLayout' else 'horizontalLayout'
+        }
+
+        with open('pl-order-blocks.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params).strip()
+        return html
+
+    elif data['panel'] == 'submission':
+        if pl.get_boolean_attrib(element, 'external-grader', False):
+            return ''
+        # render the submission panel
+        uuid = pl.get_uuid()
+        student_submission = ''
+        colour = 'badge-danger'
+        score = 0
+        feedback = None
+
+        if answer_name in data['submitted_answers']:
+            student_submission = data['submitted_answers'][answer_name]['student_raw_submission']
+        if answer_name in data['partial_scores']:
+            colour = render_html_colour(data['partial_scores'][answer_name]['score'])
+            score = data['partial_scores'][answer_name]['score'] * 100
+            feedback = data['partial_scores'][answer_name]['feedback']
+
+        html_params = {
+            'submission': True,
+            'uuid': uuid,
+            'parse-error': data['format_errors'].get(answer_name, None),
+            'student_submission': prettyPrint(student_submission),
+            'colour': colour,
+            'score': score,
+            'perfect_score': True if score == 100 else None,
+            'feedback': feedback
+        }
+
+        # Finally, render the HTML
+        with open('pl-order-blocks.mustache', 'r', encoding='utf-8') as f:
+            html = chevron.render(f, html_params).strip()
+        return html
+
+    elif data['panel'] == 'answer':
```

Comment:
I noticed this when trying another example:

![image](https://user-images.githubusercontent.com/12501485/98335792-81faaf80-1fba-11eb-9ccd-8dfb9df15783.png)

where I was using the `permutation-mode="ranking"` attribute, and asking for descending order for the correct answers.

```
<pl-order-blocks answers-name="prime-numbers-unordered" permutation-mode="ranking" header-right-column="Enter all prime numbers here:">
  <pl-answer correct="false">4</pl-answer>
  <pl-answer correct="true" ranking=2>5</pl-answer>
  <pl-answer correct="false">6</pl-answer>
  <pl-answer correct="true" ranking=1>7</pl-answer>
  <pl-answer correct="false">1</pl-answer>
  <pl-answer correct="true" ranking=4>2</pl-answer>
  <pl-answer correct="true" ranking=3>3</pl-answer>
</pl-order-blocks>
```

It looks like the grade function is taking care of the problem correctly, but the answer panel is not displaying the correct order. It actually display the same order they appear in the html.

Reply:
yeah this might be a bug, the grading function has evolved considerably, but the answer panel code hasnt been touched, will look at this

Reply:
prepare() previously implicitly assumes the user will always put `pl-answers` in ascending `ranking` order, leading to the bugged behaviour. This has been fixed.